content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql-gis.mdx

```diff
@@ -0,0 +1,183 @@
+
+## Geographic data support in Prisma
+
+At this time geographic data, and specifically PostGIS, are not supported by Prisma. Models that have geographic data columns will not have the benefit of type safety, nor will be able to use mutating operations like `create` and `update`.
+
+Prisma does support operations using `$queryRaw` and `$executeRaw` and we will be using these in order to create a Prisma Client extension with SafeQL to improve the usage of geographic data in Prisma.
+
+## 1. Set up Prisma for use with PostGIS
+
+If you haven't already, enable the `postgresqlExtensions` Preview feature and add the `postgis` PostgreSQL extension in your Prisma schema:
+
+```prisma highlight=3,8;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  extensions = [postgis]
+}
+```
+
+<Admonition type="warning">
+
+If you are not using a cloud provider, you will most likely need to install the `postgis` extension as well. [PostGIS's website](http://postgis.net/documentation/getting_started/#installing-postgis) has instructions for this, or, you can use the following Docker Compose example:
+
+```yaml
+version: '3.6'
+services:
+  pgDB:
+    image: postgis/postgis:13-3.1-alpine
+    restart: always
+    ports:
+      - '5432:5432'
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_PASSWORD: password
+      POSTGRES_DB: geoexample
+volumes:
+  db_data:
+```
+
+</Admonition>
+
+Next, create a migration using the `prisma migrate dev` command to enable the extension. The output of the migration file should look like the following:
+
+```sql
+
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+
+```
+
+Make sure that the migration is applied by running `prisma migrate status`.
+
+## 2. Create a new model that uses a geographic data column
+
+After the migration is applied, you can now add a new model that has a GIS column. For our example, we'll be using a model called `PointOfInterest`.
+
+```prisma
+model PointOfInterest {
+  id       Int                                   @id @default(autoincrement())
+  name     String
+  location Unsupported("geography(Point, 4326)")
+}
+```
+
+You'll notice that `location` is marked as `Unsupported`. This means that we lose a lot of the benefits of Prisma when working with `PointOfInterest`. We'll be using [SafeQL](https://safeql.dev/) to fix this.
+
+Run the `prisma migrate dev` command to create a migration and the table `PointOfInterest` table in your database.
+
+## 3. Integrate SafeQL
+
+SafeQL is easily integrated with Prisma. Follow [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) in order to lint `$queryRaw` and `$executeRaw` Prisma calls.
```

Comment:
Let's spell out the installation steps here instead of pointing users to the SafeQL docs.